libs/components/colorpicker/src/lib/modules/colorpicker/colorpicker.component.ts

```diff
@@ -111,218 +110,309 @@ export class SkyColorpickerComponent implements OnInit, OnDestroy {
-  public rgbaText: SkyColorpickerRgba;
-  public selectedColor: SkyColorpickerOutput;
-  public slider: SliderPosition;
-  public iconColor: string;
-  public initialColor: string;
-  public lastAppliedColor: string;
-  public isPickerVisible: boolean;
-
-  public set backgroundColorForDisplay(value: string) {
-    this._backgroundColorForDisplay = value;
+  public set backgroundColorForDisplay(value: string | undefined) {
+    this.#_backgroundColorForDisplay = value;
     if (this.pickerButtonIcon) {
-      this.iconColor = this.getAccessibleIconColor(this.selectedColor);
+      this.iconColor = this.#getAccessibleIconColor(this.selectedColor);
     }
   }
 
   public get backgroundColorForDisplay(): string {
     /* istanbul ignore next */
-    return this._backgroundColorForDisplay || '#fff';
+    return this.#_backgroundColorForDisplay || '#fff';
   }
 
-  public colorpickerId: string;
+  public set hue(change: SkyColorpickerChangeAxis) {
+    if (
+      this.#hsva &&
+      change.xCoordinate !== undefined &&
+      change.maxRange !== undefined
+    ) {
+      this.#hsva.hue = change.xCoordinate / change.maxRange;
+      this.#update();
+    }
+  }
 
-  public isOpen = false;
+  public set red(change: SkyColorpickerChangeColor) {
+    if (this.#hsva) {
+      const rgba = this.#svc.hsvaToRgba(this.#hsva);
+
+      // This code assumed non-null pre-strict mode.
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      rgba.red = change.colorValue! / change.maxRange!;
+      this.#hsva = this.#svc.rgbaToHsva(rgba);
+      this.#update();
+    }
+  }
 
-  public isVisible = true;
+  public set green(change: SkyColorpickerChangeColor) {
+    if (this.#hsva) {
+      const rgba = this.#svc.hsvaToRgba(this.#hsva);
+
+      // This code assumed non-null pre-strict mode.
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      rgba.green = change.colorValue! / change.maxRange!;
+      this.#hsva = this.#svc.rgbaToHsva(rgba);
+      this.#update();
+    }
+  }
+
+  public set blue(change: SkyColorpickerChangeColor) {
+    if (this.#hsva) {
+      const rgba = this.#svc.hsvaToRgba(this.#hsva);
+
+      // This code assumed non-null pre-strict mode.
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      rgba.blue = change.colorValue! / change.maxRange!;
+      this.#hsva = this.#svc.rgbaToHsva(rgba);
+      this.#update();
+    }
+  }
 
-  public triggerButtonId: string;
+  public set alphaAxis(change: SkyColorpickerChangeAxis) {
+    if (
+      this.#hsva &&
+      change.xCoordinate !== undefined &&
+      change.maxRange !== undefined
+    ) {
+      this.#hsva.alpha = change.xCoordinate / change.maxRange;
+      this.#update();
+    }
+  }
+
+  public set alphaColor(change: SkyColorpickerChangeColor) {
+    if (this.#hsva) {
+      // This code assumed non-null pre-strict mode.
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
☝️ Same feedback for these `eslint-disable-next-line` comments, RE: should they be TODOs?